native/react-native/common-components/button.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { StyleSheet, TouchableOpacity, Text, Image } from 'react-native';
+import PropTypes from 'prop-types';
+
+const styles = StyleSheet.create({
+  button: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    borderRadius: 4,
+    marginTop: 5,
+    marginBottom: 5
+  },
+  text: {
+    color: '#FFF',
+    fontFamily: 'oswald'
+  },
+  icon: {
+    position: 'absolute',
+    right: '85%'
+  }
+});
+
+
+ const Button = props => {
+  const { onPress, text, icon } = props;
+  let { backgroundColor, fontSize, width, height } = props;
+  if (!backgroundColor) backgroundColor = '#334B80';
+  if (!width) width = '80%';
+  if (!height) height = '8%';
+  if (!fontSize) fontSize = 20;
```

Comment:
For use cases like this, I usually like to do:
```javascript
let defaults = {
  width: '80%',
  height: '8%',
  fontSize: 20
};
let options = Object.assign({}, defaults, props);
```

I think it's a lot cleaner than having if statements all over the place.

Reply:
I really like this. I'm gonna make this change

Reply:
Even better, you can define your defaults in a StyleSheet then have the user's props override them piecemeal.  Similar to what Carlos suggests, you can just do this:

`
let defaults = {
  width: "80%",
  height: "8%",
  fontSize: 20
};
<TouchableOpacity style={{...defaults, ...props}} >
`

Reply:
It'll override the ones the user specifies as well as adding the user's additional properties at no cost to you!